test-case/verify-sof-firmware-load.sh

```diff
@@ -13,25 +13,26 @@
 ##
 
 # source from the relative path of current folder
-source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+# shellcheck source=case-lib/lib.sh
+source "$(dirname "${BASH_SOURCE[0]}")/../case-lib/lib.sh"
 
 func_opt_parse_option "$@"
 
-if [ ! "$(alias |grep 'Sub-Test')" ];then
+if alias |grep -q 'Sub-Test' ;then
+    cmd="dmesg"
+else
     # hijack DMESG_LOG_START_LINE which refer dump kernel log in exit function
     DMESG_LOG_START_LINE=$(sof-get-kernel-line.sh|tail -n 1 |awk '{print $1;}')
     cmd="sof-kernel-dump.sh"
-else
-    cmd="dmesg"
 fi
 
 dlogi "Checking SOF Firmware load info in kernel log"
-if [[ $(eval $cmd | grep "] sof-audio.*version") ]]; then
+$cmd | grep -q "sof-audio.*Firmware.*version" && {
```

Comment:
This needs to stand out as a separate commit, otherwise it's too easy to assume this is just a shellcheck cleanup like the rest of the commit.